### PR TITLE
[ATen][CUDA][CUBLAS] cublasLtMatmul increase workspace_size

### PR DESCRIPTION
**Pull request by Aidyn-A**
According to the [cuBLAS API Reference](https://docs.nvidia.com/cuda/cublas/index.html#cublassetworkspace) the recommended workspace size for Hopper is 32 MiB and for the rest architectures 4 MiB. This PR increases the workspace size accordingly. I am not aware of the recommended workspace size for HIP, that is why I am keeping it unchanged.

cc @csarofeen @ptrblck @xwang233

### CODE REVIEW

File: aten/src/ATen/cuda/CUDABlas.cpp [hunk 1]
```diff
@@ -188,8 +188,15 @@ static size_t _parseChosenWorkspaceSize() {
     // accept either env var
     val = getenv("HIPBLASLT_WORKSPACE_SIZE");
   }
+  size_t workspace_size = 1024;
+#else
+  size_t workspace_size = 4096;
+  cudaDeviceProp* p = at::cuda::getDeviceProperties(c10::cuda::current_device());
+  const int computeCapability = p->major * 10 + p->minor;
```
Comment (eqy) [inline aten/src/ATen/cuda/CUDABlas.cpp:195]: nit: can we simplify this to just `p->major >= 9`?

**Review by eqy — APPROVED** [review]
Thanks for the fix!

**Review by malfet — APPROVED** [review]
LGTM, but do you know if this might push memory utilization for some smaller GPUs beyond the limit?

**Comment by eqy** [comment]
We can add an additional qualifier that sm < 80 stick with the original 1 MiB, what do you think @Aidyn-A

**Comment by Aidyn-A** [comment]
A dilemma of perf vs memory. I guess it makes sense to decrease it for old/small GPU. Lemme commit it :+1:

**Comment by eqy** [comment]
@Aidyn-A Looks like the error is due to OOM on `DebertaForQuestionAnswering` but we might be able to WAR this by bumping the batch size divisor here : https://github.com/pytorch/pytorch/blob/eba28a6f911ad02515be0f22dc814d37b122f1cd/benchmarks/dynamo/huggingface.py#L118
🙈

**Comment by malfet** [comment]
@Aidyn-A by the way, do you know if those settings will override settings set via `CUBLAS_WORKSPACE_CONFIG`? If that is the case, it should be amended to respect user settings

**Comment by malfet** [comment]
@Aidyn-A also, you should not have merged without adding `ciflow/inductor` as you still don't get a signal whether your fixes addressed the problem or not

**Comment by Aidyn-A** [comment]
@malfet, I apologize for rushing this PR to merge as I was not aware of neither of the `ciflow/inductor`. 

As for `CUBLAS_WORKSPACE_CONFIG`, it will not override the settings, as `cublasLtMatmul` relies on supplied workspace only, unlike the classic cuBLAS ops that depend on `CUBLAS_WORKSPACE_CONFIG` or `cublasSetWorkspace`.

**Comment by mortzur** [comment]
A workflow that sets:
os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
[for determinism](https://docs.nvidia.com/cuda/cublas/index.html#results-reproducibility), fails following this change.
Is it possible to check if CUBLAS_WORKSPACE_CONFIG if provided by the user and enforce it if so?

**Comment by ptrblck** [comment]
@mortzur Could you describe what exactly is failing? This PR changes the **cublasLt** workspace and does not touch **cublas** workspace handling.

**Comment by mortzur** [comment]
[ptrblck](https://github.com/ptrblck) you're right, I can set CUBLASLT_WORKSPACE_SIZE to fix the workspace size. no issue here.

**Comment by ptrblck** [comment]
@mortzur Great, thanks for confirming!

@malfet is this solving the internal test issues and is anything else needed to land the PR?

**Pull request closed without merging**